### PR TITLE
Add gitflow documentation

### PR DESCRIPTION
**Pull request by mfcastellani**
### What does this PR does?

This PR adds a document with the flow for using git.

### CODE REVIEW

File: council/gitflow.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+the next big release.
+
+The key moment to branch off a new release branch from develop is when develop (almost) 
+reflects the desired state of the new release. At least all features that are targeted 
+for the release-to-be-built must be merged in to develop at this point in time. All 
+features targeted at future releases may not—they must wait until after the release 
+branch is branched off.
+
+It is exactly at the start of a release branch that the upcoming release gets assigned 
+a version number—not any earlier. Up until that moment, the develop branch reflected
```
Comment (arnaubennassar) [inline council/gitflow.md:211]: I'd specify that the version number will be sufixed with a `-rcX`, example: `v1.2.0-rc5`. Subsequent commits will increase the X in `-rcX`, although maybe not all commits must include the tag. At the end of the day those tags are used to indicate the QA team that "this code should be tested to check if it's production ready"

File: council/gitflow.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+$ git pull origin develop
+
+# create release branch
+$ git checkout -b release/name-of-release
+Switched to a new branch "release/name-of-release"
+```
+
+After that open a PR pointing to `master` to put the release branch inside the 
+approval flow. After approved the responsible for open the PR must merge into `master`,
+delete the release branch and **tag new version**.
```
Comment (arnaubennassar) [inline council/gitflow.md:241]: At the moment, `master` is protected in a way that only some members can add commits. So I'd rephrase this a bit because it gives the sensation that "anyone can do it". Since the repo is public this protection is a must.

**Review by arnaubennassar — CHANGES_REQUESTED** [review]

**Review by jeffprestes — APPROVED** [review]

File: council/gitflow.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+Each developer pulls and pushes to origin. But besides the centralized push-pull 
+relationships, each developer may also pull changes from other peers to form 
+sub teams. For example, this might be useful to work together with two or more 
+developers on a big new feature, before pushing the work in progress to origin 
+prematurely. In the figure above, there are subteams of Alice and Bob, Alice and 
+David, and Clair and David.
+
+Technically, this means nothing more than that Alice has defined a Git remote, 
+named bob, pointing to Bob’s repository, and vice versa.
```
[the remark this answers is not in the document]
Comment (jeffprestes) [reply council/gitflow.md:39]: IMHO, information is never too much. Also, have this described helps people do not know Gitflow process.

File: council/gitflow.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+After merge with *master* you need to update the *develop* branch with the hotfix too
+to avoid conflicts and loose code. 
+
+```bash
+$ git checkout develop
+Switched to branch 'develop'
+
+$ git merge --no-ff hotfix/1.2.1
+Merge made by recursive.
+(Summary of changes)
```
[the remark this answers is not in the document]
Comment (jeffprestes) [reply council/gitflow.md:188]: @arnaubennassar could you please give us example of commands you think are better?

**Review by jeffprestes — APPROVED** [review]

File: council/gitflow.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (arnaubennassar) [reply council/gitflow.md]: *assuming that `hotfix/doom` is a branch that was created from master, added a few commits and merged to master*

1. Create PR to merge `hotfix/doom` into `develop` (using GitHub web)
2. Fix conflicts (locally or using GitHub web), commit and push changes
3. Review, aprove and merge (using GitHub web)

**Review by arnaubennassar — APPROVED** [review]
